Source/ARTRealtime.m

```diff
@@ -773,14 +773,17 @@ - (void)onConnected:(ARTProtocolMessage *)message {
     
     // Resuming
     if (_resuming) {
-        if (![message.connectionId isEqualToString:self.connection.id_nosync]) { // RTN15c3
-            [self.logger warn:@"RT:%p connection \"%@\" has reconnected, but resume failed. Reattaching any attached channels", self, message.connectionId];
-            // Reattach all channels
+        if ([message.connectionId isEqualToString:self.connection.id_nosync]) { // RTN15c6
+            [self.logger debug:@"RT:%p connection \"%@\" has reconnected and resumed successfully", self, message.connectionId];
+            // Reattach all channels with specified states
             for (ARTRealtimeChannelInternal *channel in self.channels.nosyncIterable) {
-                [channel reattachWithReason:message.error callback:nil];
+                if (channel.state == ARTRealtimeChannelAttaching || channel.state == ARTRealtimeChannelAttached || channel.state == ARTRealtimeChannelSuspended) {
```

Comment:
Check on states was inside `reattachWithReason:`. Maybe you removed it later, but I'll leave this comment to not seek for this commit again later.